lib/curl_setup.h

```diff
@@ -778,9 +778,15 @@ endings either CRLF or LF so 't' is appropriate.
 #  endif
 #endif /* DONT_USE_RECV_BEFORE_SEND_WORKAROUND */
 
-/* for systems that don't detect this in configure, use a sensible default */
+/* for systems that don't detect this in configure */
 #ifndef CURL_SA_FAMILY_T
-#define CURL_SA_FAMILY_T unsigned short
+#  if defined (__APPLE__)
+/* as type-defined in the sys/_types/_sa_family_t.h file */
+#    define CURL_SA_FAMILY_T sa_family_t
+#  else
+/* use a sensible default */
+#    define CURL_SA_FAMILY_T unsigned short
+#  endif
```

Comment:
This is better than before, but the configure script version actually test-compiles to make sure that it is valid on the particular platform. Shouldn't this too? Some/many platforms use the type `ADDRESS_FAMILY` which should be preferred if available!

Reply:
Totally agreed! I surely have struggled as well for I wanted to avoid implementing the same logic in all places (`configure.ac`, `CMakeLists.txt`, and this file).

One specific problem for `ADDRESS_FAMILY` is that the symbol might as well be a typedef than macro: https://github.com/tpn/winsdk-10/blob/9b69fd26ac0c7d0b83d378dba01080e93349c2ed/Include/10.0.10240.0/shared/ws2def.h#L76-L83 thus might not be easily handled at pre-processing time. :(